src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java

```diff
@@ -223,9 +227,26 @@ ComponentRecipe getPackageRecipe(@NonNull ComponentIdentifier pkgId) throws Pack
      * @param compId component identifier
      * @throws PackageLoadingException if deletion of the component failed
      */
-    void deleteComponent(@NonNull ComponentIdentifier compId) throws PackageLoadingException {
+    void deleteComponent(@NonNull ComponentIdentifier compId, @NonNull ArtifactDownloaderFactory artifactDownloaderFactory) throws PackageLoadingException, InvalidArtifactUriException {
         logger.atDebug("delete-component-start").kv("componentIdentifier", compId).log();
         IOException exception = null;
+        // issues #1111
+        // delete docker image before removing the recipe file
+        Optional<String> componentRecipeContent = findComponentRecipeContent(compId);
+        if (componentRecipeContent.isPresent()) {
+            // get recipe, all recipes are saved as yml files
+            ComponentRecipe recipe = getPackageRecipe(compId);
+            Path packageArtifactDirectory = resolveArtifactDirectoryPath(compId);
+            for (ComponentArtifact artifact : recipe.getArtifacts()) {
+                ArtifactDownloader downloader = artifactDownloaderFactory
+                        .getArtifactDownloader(compId, artifact, packageArtifactDirectory);
+                try {
+                    downloader.cleanup(this);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
```

Comment:
Absolutely do not re-throw the exception, this will cause wide-spread impact.

Look at how the existing logic handles exceptions. They will check

```
if (exception == null) {
                exception = e;
            } else {
                exception.addSuppressed(e);
            }
```